product_variant_sale_price/hooks.py

```diff
@@ -5,9 +5,22 @@
 
 def set_sale_price_on_variant(cr, registry):
     sql = """
-            UPDATE product_product pp
-            SET fix_price = (SELECT list_price
-                                FROM product_template pt
-                                WHERE pp.product_tmpl_id = pt.id)
+        UPDATE product_product pp
```

Comment:
You can check that this works making a test where you force the initial conditions on test records (perhaps making a SQL for emptying fix_price column), and then calling this method to see if it has been correctly filled. You can improve even the method to admit a keyword argument to limit the query to one record, and thus the test will be faster.